### PR TITLE
Fix handling of LOH and POH in allocation helpers

### PR DESCRIPTION
**Pull request by jkotas**

### CODE REVIEW

File: src/coreclr/src/nativeaot/Runtime/GCHelpers.cpp [hunk 1]
```diff
@@ -397,7 +397,7 @@ static Array* AllocateNewArrayImpl(Thread* pThread, EEType* pArrayEEType, uint32
         }
     }
 
-    if (size > RH_LARGE_OBJECT_SIZE)
+    if (size >= RH_LARGE_OBJECT_SIZE)
```
Comment (jkotas) [inline src/coreclr/src/nativeaot/Runtime/GCHelpers.cpp:400]: Has to match assembly helpers.

File: src/coreclr/src/nativeaot/Runtime/GCHelpers.cpp [hunk 2]
```diff
@@ -412,7 +412,7 @@ static Array* AllocateNewArrayImpl(Thread* pThread, EEType* pArrayEEType, uint32
     pArray->set_EEType(pArrayEEType);
     pArray->InitArrayLength(numElements);
 
-    if (size >= RH_LARGE_OBJECT_SIZE)
+    if (flags & GC_ALLOC_USER_OLD_HEAP)
```
Comment (jkotas) [inline src/coreclr/src/nativeaot/Runtime/GCHelpers.cpp:415]: Matches CoreCLR

**Comment by jkotas** [comment]
System.Runtime.Tests are running till completion with these changes

**Review by MichalStrehovsky — APPROVED** [review]
Sigh. That's my fault. Thank you!

**Pull request merged**